### PR TITLE
Catalog

### LINKED ISSUE #9: Update auction page with actual catalog details
As of now auction page shows just a number on which people bid. Change that with actual catalog details by fetching it from backend. This requires a bit of change to socket code too.

### PR DESCRIPTION
**Pull request by shakti97**
This pr will solve and close #9 and #14 issue

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -32,28 +32,39 @@ app.get('/*', function(req, res) {
...
 //socket server
 const io = require('socket.io')(server, {wsEngine: 'ws'});
 
 const handleConnections = require('./src/socketRoutes/connectionManager');
+const handleBiding = require('./src/socketRoutes/bidManager');
 
 //socket routes
 io.sockets.on('connection', socket => {
     socket.on('openAuction', (namespace, owner_id) => {
+        console.log('open Auction');
```
Comment (hariprasanths) [inline app.js]: Remove the debug logs

File: assets/app/jsx/AdminPanel.jsx [hunk 2]
```diff
@@ -87,10 +93,27 @@ class AdminPanel extends Component {
                 });
             })
             .catch(err => {
-                console.log(err);
+                notifyError(err.response);
             });
     }
-
+    getCatalog = () => {
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx]: argument passed during function call
should be 
getCatalog = (data) => {

File: assets/app/jsx/AdminPanel.jsx [hunk 3]
```diff
@@ -87,10 +93,27 @@ class AdminPanel extends Component {
...
+        if (this.state.is_open) {
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:107]: setState is asyncronous, so this check will mostly fail before this.state.is_open is changed to true
use the value of data.is_open for the check

File: assets/app/jsx/AdminPanel.jsx [hunk 4]
```diff
@@ -87,10 +93,27 @@ class AdminPanel extends Component {
...
+                    } else {
+                        notifyError(response.message);
+                    }
+                })
+                .catch(err => {
+                    notifyError(err.response);
+                });
+        }
+    };
     openAuction() {
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:123]: this function needs to be called on init if the auction is already opened

File: webpack.config.js [hunk 5]
```diff
@@ -1,56 +1,62 @@
-const webpack = require('webpack');
-const AssetsPlugin = require('assets-webpack-plugin');
-const UglifyJSPlugin = require('uglifyjs-webpack-plugin');
+const webpack = require("webpack");
+const AssetsPlugin = require("assets-webpack-plugin");
+const UglifyJSPlugin = require("uglifyjs-webpack-plugin");
```
Comment (hariprasanths) [inline webpack.config.js]: Whole file needs to be formatted
Indentation and single quotes

**Review by hariprasanths — CHANGES_REQUESTED** [review]

File: assets/app/jsx/AdminPanel.jsx [hunk 6]
```diff
@@ -87,35 +100,86 @@ class AdminPanel extends Component {
                 });
             })
             .catch(err => {
-                console.log(err);
+                notifyError(err.response);
             });
     }
-
+    getCatalog = data => {
+        if (this.state.is_open) {
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:107]: Use data.is_open for the check

Comment (shakti2379) [reply assets/app/jsx/AdminPanel.jsx:107]: now its resolved, because I m calling this function in the setstate callback

**Review by hariprasanths — APPROVED** [review]

**Pull request merged**